### ISSUE TITLE
Use public key cryptography instead of AES

### ISSUE THREAD

**Issue by Mgayar-sci** [Youssef-S-Negm/Bubbles#73]
Create a public/private key pair on user creation, 
`user1` has `pu1`(public key)/`pv1` and `user2` has `pu2`/`pv2`
When `user1` wants to send a message to `user2`, he encrypts the message using `pu2`
When `user2` wants to read any message, he decrypts it using `pv2`
You can add another step by encrypting message from `user1` to `user2` using `pv1` then encrypt the result using `pu2`
`C=Encrypt(Encrypt(M,pv1),pu2)`
On the other side `user2` should do a double decryption as well
`M=Decrypt(Decrypt(C,pv2),pu1)`
This insures that the message was sent by `user1` not by any other user.

**Comment by Youssef-S-Negm** [comment]
I have a question regarding public and private keys. Should the public key be stored in the user entity in the database, and the private key be stored locally?

**Comment by Mgayar-sci** [comment]
> I have a question regarding public and private keys. Should the public key be stored in the user entity in the database, and the private key be stored locally?

Yes, that's why when you change your device, you get a new pair of keys. Make sure to save the device id somewhere as well.
